MazerPlatformer/GameWorld.cs

```diff
@@ -94,30 +95,31 @@ internal void LoadContent(int levelNumber, int? overridePlayerHealth = null, int
             _rooms = _level.GetRooms();
 
             _removeWallTimer.Start();
+            return Nothing;
 
         }
 
-        private void AddToGameObjects(Dictionary<string, GameObject> levelGameObjects)
-        {
-            foreach (var levelGameObject in levelGameObjects)
-                AddToGameObjects(levelGameObject.Key, levelGameObject.Value);
-        }
+        private Either<IFailure, Unit> AddToGameObjects(Dictionary<string, GameObject> levelGameObjects)
+            => levelGameObjects
+                .Select(levelGameObject => AddToGameObjects(levelGameObject.Key, levelGameObject.Value))
+                .AggregateFailures();
```

Comment:
Aggregating multiple failures is inherently a problem because you want to report one failure to indicate that failures occurred but you've got multiple fails to condense into one.